user_tools/src/spark_rapids_tools/tools/core/qual_output_reader.py

```diff
@@ -0,0 +1,248 @@
+                                if result.success:
+                                    app_df = result.df
+                                    # Add App ID column as the first column to maintain conformity
+                                    app_df.insert(0, 'App ID', app_id)
+                                    combined_dfs.append(app_df)
+                        except Exception:  # pylint: disable=broad-except
+                            continue  # Skip files that can't be read
+
+            # Combine all DataFrames
+            if combined_dfs:
+                return pd.concat(combined_dfs, ignore_index=True)
+            return pd.DataFrame()
+
+        except Exception:  # pylint: disable=broad-except
+            return pd.DataFrame()
+
+    def read_raw_metric_per_app_files(self,
+                                      file_name: str,
+                                      read_csv_kwargs: Optional[dict] = None) -> Dict[str, pd.DataFrame]:
+        """
+        Read a specific raw metric file from each application's raw_metrics directory.
+
+        Args:
+            file_name: Name of the metric file to read from each application's raw_metrics folder
+            read_csv_kwargs: Optional dictionary of arguments to pass to pandas read_csv
+
+        Returns:
+            Dictionary where keys are application IDs and values are DataFrames containing
+            the corresponding application's metrics data
+        """
+        metrics = {}
+        try:
+            qual_core_path = FSUtil.build_path(self.output_directory, 'qual_core_output')
+            raw_metrics_path = FSUtil.build_path(qual_core_path, 'raw_metrics')
+
+            if not Path(raw_metrics_path).exists():
+                return metrics
+
+            raw_metrics_dir = Path(raw_metrics_path)
+
+            for app_dir in raw_metrics_dir.iterdir():
+                if app_dir.is_dir():
+                    app_id_str = app_dir.name  # Extract App ID from directory name
+                    metric_file_path = app_dir / file_name
+
+                    try:
+                        if metric_file_path.exists():
+                            result = DataUtils.load_pd_df(str(metric_file_path), read_csv_kwargs=read_csv_kwargs)
+                            if result.success:
+                                metrics[app_id_str] = result.df
+                            else:
+                                metrics[app_id_str] = pd.DataFrame()
+                        else:
+                            metrics[app_id_str] = pd.DataFrame()
+                    except Exception:  # pylint: disable=broad-except
+                        # Some apps may not have the given metrics file, maintain consistency
+                        metrics[app_id_str] = pd.DataFrame()
+
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        return metrics
+
+    def _get_cluster_info_json_to_csv_mapping(self) -> dict:
```

Comment:
While I understand the motive to add this method, it takes us back to what is the CSV file is needed for?
As per our offline discussion, the clusterInfo is a structured data, and it unstable because of the frequent changes to the AutoTuner. This is why the CSV file format was removed from the core-tools.
Therefore, the question is: "_why did this PR re-create the CSV file in the user-tools ?_"